app/lib/authorizer.rb

```diff
@@ -0,0 +1,113 @@
+# This class defines authorization rules.
+# If you want student data, ask through this class.
+# Ideally, start with using `authorized_only`.
+# Drop down to calling individual methods if you need to.
+#
+# This class should be fully tested, and all changes should be reviewed.
+# 
+# Controller usage:
+#   student = authorized_only { Student.find(params[:id]) }
+#   homeroom = authorized_only { Homeroom.(params[:id]) }
+#   students = authorized_only { homeroom.students } 
+#
+# Standalone usage:
+#   authorizer = Authorizer.new(educator)
+#   student = authorizer.authorized_only { Student.find(params[:id]) }
+#   homeroom = authorizer.authorized_only { Homeroom.(params[:id]) }
+#   students = authorizer.authorized_only { homeroom.students } 
+#
+# Instead of:
+#   Homeroom.find(id).students # no authorization check
+#   Section.find(id).students # no authorization check
+#   current_educator.homeroom.students # doesn't consider other roles
```

Comment:
@kevinrobinson What do you mean by "doesn't consider other roles"?

Reply:
In other words, for a given homeroom page, this doesn't do what you'd want for all user types (eg, district admin).

Reply:
Ooooooh right, that makes sense. For example, if `current_educator` is a district admin s/he won't be assigned a `homeroom`.